### ISSUE TITLE
Feature request: Support instantiation of multiple julia environments in parallel

### ISSUE THREAD

**Issue by schlichtanders** [JuliaLang/Pkg.jl#3495]
With Julia 1.9 it gets more important to do proper precompilation ahead of time. I am using Pluto and like to instantiate all notebook environments with high speed.

Currently, when just spawning multiple julia processes and instantiating several environments, I get a GitError as soon as two environments have the same dependency (there is probably some randomness involved, but it failes constantly for me - there are several overlaps)

```julia
GitError(Code:ERROR, Class:OS, could not open '/home/myhome/.julia/packages/MyPackage/1opWe/Project.toml' for writing: Permission denied)
```

It would be really great to have a performant way of instantiating multiple environments at once.

~~EDIT: Just realized, that I only tested this Julia 1.8.5, I am going to test on Julia 1.9 immediately~~ Applies to 1.9 as well

**Comment by schlichtanders** [comment]
@IanButterworth  would this also be solved by [the mentioned issue](https://github.com/JuliaLang/julia/pull/49052)

**Comment by vchuravy** [comment]
Not quite. Julia 1.10 will be better around races on cache-files, but there is still an open question around two separate processes calling `instantiate` concurrently, on the same project or a project that contains the same dependency.

**Comment by IanButterworth** [comment]
We should fix this. Specifically the race around https://github.com/JuliaLang/Pkg.jl/blob/116ba910c74ab565d348aa8a50d6dd10148f11ab/src/Operations.jl#L1080-L1083
